Fix golint warnings

srderson opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
## Description

<!--- Describe your changes in detail. -->

This PR fixes numerous `golint` warnings. There are no functional changes.
## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

<!--- If it fixes an open issue, please link to the issue here. -->

Developers are now being asked to run `golint` on all proposed changes and resolve any warnings. This change fixes pre-existing warnings.
## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->

<!--- If this PR does not contain a new test case, explain why. -->

I have run `make checks`. There are no new tests as this PR only resolves `golint` warnings and does not add any new functionality.
## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff)
- [x] Either no new documentation is required by this change, OR I added new documentation
- [x] Either no new tests are required by this change, OR I added new tests
- [x] I have run [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports), [go vet](https://golang.org/cmd/vet/), and [golint](https://github.com/golang/lint). I have cleaned up all valid errors and warnings in code I have added or modified. These tools may generate false positives. Don't be worried about ignoring some errors or warnings. The goal is clean, consistent, and readable code.

Signed-off-by: Sheehan Anderson sheehan@us.ibm.com

srderson committed:
Fix golint warnings

srderson committed:
Fix golint warnings

corecode commented on core/ledger/statemgmt/trie/trie_node.go:
```diff
@@ -134,7 +134,7 @@ func (trieNode *trieNode) marshal() ([]byte, error) {
 
 	// write value marker explicitly because rocksdb apis convertes a nil into an empty array and protobuf does it other-way around
-	var valueMarker uint64 = 0
+	var valueMarker uint64 // default value is 0
```
Now this is a lint warning that I would have ignored.  The assignment is more clear than the comment.

ghaskins replied:
I agree with @corecode

christo4ferris commented:
I agree, and think that a comment noting the lint exception should be added so we don't get this multiple times

srderson committed:
Ignore 'drop = 0' golint warning

srderson commented:
@corecode @ghaskins Agree with comments about dropping '= 0' making the assignment less clear. I have reverted the two cases where I did that and added a comment about ignoring the warning per @christo4ferris's suggestion. Thanks.

ghaskins commented:
LGTM

Reviewed-by: Greg Haskins gregory.haskins@gmail.com
